lib/core/src/Cardano/Wallet.hs

```diff
@@ -535,14 +545,15 @@ newWalletLayer tracer block0 feePolicy db nw tl = do
         -> WalletId
         -> BlockHeader
         -> IO ()
-    restoreSleep t wid slot = do
-        let tenSeconds = 10000000 in threadDelay tenSeconds
-        runExceptT (networkTip nw) >>= \case
-            Left e -> do
-                logError t $ "Failed to get network tip: " +|| e ||+ ""
-                restoreSleep t wid slot
-            Right tip ->
-                restoreStep t wid (slot, tip)
+    restoreSleep t wid slot =
+        let halfSlotLengthDelay = fromIntegral (diffTimeToPicoseconds slotLength `div` 2000000) in do
```

Comment:
Minor point: the `let...in...do` notation is rather awkward. 
Instead, a top-level `do` works better:

```hs
restoreSleep t wid slot = do
    let ...
    threadDelay ...
```

Reply:
corrected